### PR TITLE
Added watch option to build and serve command

### PR DESCRIPTION
**Pull request by dylanrodr-heig**

### CODE REVIEW

**Review by clemdiep-heig — APPROVED** [review]
Seems great ! 🍻

**Review by clemdiep-heig — APPROVED** [review]
Oupsi, forgot to test to modify the templates.

It's doesn't rebuild the site pages. The problem seems to be related to watching in a sub-folder, I tested with a file `pages/page1.md` too and when it's modified no rebuild is done.

**Review by clemdiep-heig — CHANGES_REQUESTED** [review]
Wrong approvement with previous comment...

Thanks GitHub to not let me delete my message :)

**Comment by dylanrodr-heig** [comment]
@clemdiep-heig corrected :D

File: src/main/java/ch/heigvd/statique/utils/Watcher.java [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+                    path = parentPath.resolve(path);
+
+                    try {
+                        registerDir(path);
+                    } catch (IOException e) {
+                        e.printStackTrace();
+                    }
+                }
+
+                if (ENTRY_CREATE.equals(kind)) {
```
Comment (clemdiep-heig) [inline src/main/java/ch/heigvd/statique/utils/Watcher.java:113]: Couldn't you merge those two if together ?

```suggestion
                if (ENTRY_CREATE.equals(kind)) {
                    //this is not a complete path
                    Path path = ev.context();
                    //need to get parent path
                    Path parentPath = keyPathMap.get(key);
                    //get complete path
                    path = parentPath.resolve(path);

                    try {
                        registerDir(path);
                    } catch (IOException e) {
                        e.printStackTrace();
                    }
```

**Review by clemdiep-heig — APPROVED** [review]
Great sub folder are working now !

Just a little comment

File: src/main/java/ch/heigvd/statique/utils/Watcher.java (hunk 1 above)
Comment (dylanrodr-heig) [reply src/main/java/ch/heigvd/statique/utils/Watcher.java]: Yes, you're right ! Thx ^^

**Comment by dylanrodr-heig** [comment]
@clemdiep-heig Requested changes done ^^

**Review by clemdiep-heig — APPROVED** [review]

**Review by Isotech42 — APPROVED** [review]
Wonderful work !

**Pull request merged**